packages/synthetix-main/contracts/storage/FundVaultStorage.sol

```diff
@@ -31,6 +33,27 @@ contract FundVaultStorage is IVaultModuleStorage {
         mapping(uint256 => SetUtil.Bytes32Set) liquidityItemsByAccount;
         /// @dev minted USD
         mapping(uint256 => uint256) usdByAccount;
+        /// @dev rewards
+        RewardDistribution[] rewards;
+    }
+
+    struct RewardDistribution {
```

Comment:
Rewards Distribution can be moved to a specific module/storage pair and if something is needed from the vault (or viceversa) use mixins in order to maintain single responsibilities on modules (and more important) storage